p2p/event_handler.go

```diff
@@ -40,6 +41,30 @@ func (m *p2pModule) HandleEvent(event *anypb.Any) error {
 			}
 		}
 
+	case messaging.StateMachineTransitionEventType:
+		stateMachineTransitionEvent, ok := evt.(*messaging.StateMachineTransitionEvent)
+		if !ok {
+			return fmt.Errorf("failed to cast event to StateMachineTransitionEvent")
+		}
+
+		if stateMachineTransitionEvent.NewState == string(coreTypes.StateMachineState_P2P_Bootstrapping) {
+			addrBook := m.network.GetAddrBook()
+			if len(addrBook) == 0 {
+				m.logger.Warn().Msg("No peers in addrbook, bootstrapping")
+
+				if err := bootstrap(m); err != nil {
+					return err
+				}
+			}
+			if !isSelfInAddrBook(m.address, addrBook) {
+				m.logger.Warn().Msg("Self address not found in addresbook, advertising")
```

Comment:
https://github.com/pokt-network/pocket/pull/491/files#r1108018541
> WHy do we only advertise if we're not in the address book?

Good question. It's because the nodes are created with a genesis and if you are node 005+ you have the first 4 but not self. It's something we need to figure out.

Reply:
1. We might end up moving the addr book bootstrapping to a config file rather than genesis.
2. Unrelated to (1), there's no harm to advertise even if some part of the network is already aware of your existence

p.logger.Info().Bool("TODO", true).Msg("Advertise self to network")

cc @bryanchriswhite

Reply:
Agreed